{"seeAlsoSections":[{"title":"Errors","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError"],"generated":true}],"schemaVersion":{"major":0,"minor":3,"patch":0},"metadata":{"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSResponseError"}],"title":"AWSResponseError","roleHeading":"Structure","role":"symbol","symbolKind":"struct","externalID":"s:8SotoCore16AWSResponseErrorV","modules":[{"name":"SotoCore"}],"navigatorTitle":[{"kind":"identifier","text":"AWSResponseError"}]},"abstract":[{"type":"text","text":"Response Error type returned by Soto if the error code is unrecognised"}],"sections":[],"identifier":{"url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","interfaceLanguage":"swift"},"relationshipsSections":[{"identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP","doc:\/\/codes.soto.soto-core\/s5ErrorP","doc:\/\/codes.soto.soto-core\/s8SendableP"],"kind":"relationships","title":"Conforms To","type":"conformsTo"}],"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore"]]},"topicSections":[{"title":"Initializers","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/init(errorCode:context:)"]},{"title":"Instance Properties","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/context","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/description","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/errorCode"]},{"title":"Default Implementations","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/AWSErrorType-Implementations","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/Error-Implementations"],"generated":true}],"variants":[{"paths":["\/documentation\/sotocore\/awsresponseerror"],"traits":[{"interfaceLanguage":"swift"}]}],"kind":"symbol","primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSResponseError"}],"languages":["swift"],"platforms":["macOS"]}]}],"references":{"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError/init(errorCode:context:)":{"role":"symbol","title":"init(errorCode:context:)","fragments":[{"kind":"identifier","text":"init"},{"kind":"text","text":"("},{"kind":"externalParam","text":"errorCode"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"kind":"text","text":", "},{"kind":"externalParam","text":"context"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"AWSErrorContext","preciseIdentifier":"s:8SotoCore15AWSErrorContextV"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"initialize error"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/init(errorCode:context:)","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsresponseerror\/init(errorcode:context:)"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSClientError":{"role":"symbol","title":"AWSClientError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSClientError"}],"abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSClientError"}],"url":"\/documentation\/sotocore\/awsclienterror"},"doc://codes.soto.soto-core/s23CustomStringConvertibleP":{"type":"unresolvable","title":"Swift.CustomStringConvertible","identifier":"doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError":{"role":"symbol","title":"AWSServerError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSServerError"}],"abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSServerError"}],"url":"\/documentation\/sotocore\/awsservererror"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError":{"role":"symbol","title":"AWSResponseError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSResponseError"}],"abstract":[{"type":"text","text":"Response Error type returned by Soto if the error code is unrecognised"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSResponseError"}],"url":"\/documentation\/sotocore\/awsresponseerror"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError/description":{"role":"symbol","title":"description","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"description"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/description","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsresponseerror\/description"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSRawError":{"role":"symbol","title":"AWSRawError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSRawError"}],"abstract":[{"type":"text","text":"Raw unprocessed error."}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSRawError"}],"url":"\/documentation\/sotocore\/awsrawerror"},"doc://codes.soto.soto-core/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/codes.soto.soto-core\/s8SendableP"},"doc://codes.soto.soto-core/s5ErrorP":{"type":"unresolvable","title":"Swift.Error","identifier":"doc:\/\/codes.soto.soto-core\/s5ErrorP"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorContext":{"role":"symbol","title":"AWSErrorContext","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSErrorContext"}],"abstract":[{"type":"text","text":"Additional information about error"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSErrorContext"}],"url":"\/documentation\/sotocore\/awserrorcontext"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError/AWSErrorType-Implementations":{"role":"collectionGroup","title":"AWSErrorType Implementations","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/AWSErrorType-Implementations","kind":"article","type":"topic","url":"\/documentation\/sotocore\/awsresponseerror\/awserrortype-implementations"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError/context":{"role":"symbol","title":"context","fragments":[{"kind":"keyword","text":"let"},{"kind":"text","text":" "},{"kind":"identifier","text":"context"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"AWSErrorContext","preciseIdentifier":"s:8SotoCore15AWSErrorContextV"},{"kind":"text","text":"?"}],"abstract":[{"type":"text","text":"additional context information related to the error"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/context","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsresponseerror\/context"},"doc://codes.soto.soto-core/documentation/SotoCore":{"role":"collection","title":"SotoCore","abstract":[{"type":"text","text":"The core framework for Soto, a Swift SDK for AWS"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","kind":"symbol","type":"topic","url":"\/documentation\/sotocore"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError/errorCode":{"role":"symbol","title":"errorCode","fragments":[{"kind":"keyword","text":"let"},{"kind":"text","text":" "},{"kind":"identifier","text":"errorCode"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"}],"abstract":[{"type":"text","text":"Error code return by AWS"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/errorCode","kind":"symbol","type":"topic","url":"\/documentation\/sotocore\/awsresponseerror\/errorcode"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError/Error-Implementations":{"role":"collectionGroup","title":"Error Implementations","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError\/Error-Implementations","kind":"article","type":"topic","url":"\/documentation\/sotocore\/awsresponseerror\/error-implementations"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType":{"role":"symbol","title":"AWSErrorType","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSErrorType"}],"abstract":[{"type":"text","text":"Standard Error type returned by Soto."}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AWSErrorType"}],"url":"\/documentation\/sotocore\/awserrortype"}}}